frontend/array_cmath/include/algebra/array_cmath.hpp

```diff
@@ -17,6 +17,8 @@
 using algebra::cmath::operator*;
 using algebra::cmath::operator-;
 using algebra::cmath::operator+;
+using algebra::cmath::inverse;
+using algebra::cmath::transpose;
```

Comment:
You should not pull "math specific" functions into the global namespace. The operators are pulled into the global namespace to make it possible to use statements like:

```c++
auto array = ...;
float multiplier = ...;
auto multiplied_array = array * multiplier;
```

outside of the namespace in which the operator was defined. Since for a given storage type there's pretty much just a single way of how to perform those very low level operations, having multiple implementations of the operators sharing the global namespace should be okay. Overloading rules should work correctly for them, even when using two different "front-ends" inside of the same source file.

But for more complex operations I would prefer if we could allow clients to choose their implementation through template arguments. Kind of like what you did with the `matrix_generator` type. I think it would be nice to try to implement all operators like that. That clients would have to use some template class/struct, whose functions they call. Because in such a setup it should be pretty easy to write code on top of this mathematics in such an abstract way that the user could "template" the classes that we would write on top of these classes. (And we wouldn't have to make a decision at compile time of which one front-end we would use in the project.)